src/install.rs

```diff
@@ -136,3 +154,76 @@ fn chmod_x(file: &PathBuf) {
         error!("Failed to run chmod +x: {}", e)
     };
 }
+
+fn mac_process_zip(archive_path: &PathBuf) -> Result<()> {
+    // Using Mac unzip because it keeps the execution flags intact.
+    let archive_parent = archive_path.parent().with_context(|| {
+        format!(
+            "Unable to determine parent from {}",
+            archive_path.to_string_lossy()
+        )
+    })?;
+    if let Err(e) = Command::new("/usr/bin/unzip")
+        .arg(archive_path.to_string_lossy().to_string())
+        .arg("-d")
+        .arg(archive_parent.to_string_lossy().to_string())
+        .output()
+    {
+        return Err(anyhow!(
+            "Unzip of {} to {} failed! {}",
+            archive_path.to_string_lossy(),
+            archive_parent.to_string_lossy(),
+            e
+        ));
+    }
+
+    // delete the zip file - in this case the version is usuable, therefore only log message
+    if let Err(e) = fs::remove_file(archive_path) {
+        error!("Deletion of archive file failed! {}", e);
+    }
+    Ok(())
+}
+
+fn mac_process_dmg(archive_path: &PathBuf) -> Result<()> {
+    // Mount the disk image file
+    let attach_info = dmg::Attach::new(archive_path)
+        .attach()
+        .with_context(|| format!("Mounting the dmg file failed"))?;
+
+    // Copy the application (which is in fact a directory)
+    let mut app_source_path = PathBuf::from("/Volumes");
+    let stem = archive_path.file_stem().with_context(|| {
+        format!(
+            "Unable to determine stem from {}",
+            archive_path.to_string_lossy()
+        )
+    })?;
+    app_source_path.push(stem);
+    app_source_path.push("Endless Sky.app");
+    let parent = archive_path.parent().with_context(|| {
+        format!(
+            "Unable to determine parent from {}",
+            archive_path.to_string_lossy()
+        )
+    })?;
+    let app_target_path = PathBuf::from(parent);
+    let mut options = CopyOptions::new();
+    options.overwrite = true;
+    if let Err(e) = copy(app_source_path.clone(), app_target_path.clone(), &options) {
```

Comment:
It may be possible to pass by reference here.
```suggestion
    if let Err(e) = copy(&app_source_path, &app_target_path, &options) {
```

Reply:
Adjusted